### PR TITLE
Add bazel actions workflow

### PR DESCRIPTION
**Pull request by rsemenov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61040
* **#61039**

- Added a new template for bazel GH Actions workflow
- Simplified the workflow based on @malfet's suggestion by combining build and test jobs into one as we only run a small subset of tests for bazel
- Tested the run to make sure it succeeds
- Build step takes 4 minutes, test step takes 7 minutes

The downside of this approach is that I duplicated some of the jobs in a new template file.  Alternative solution would be to use something like this https://jinja.palletsprojects.com/en/3.0.x/templates/#template-inheritance, however, that is better to be done in a separate PR as linux and windows workflows would need to be changed. Another solution is to use a bunch of if else statements in a linux workflow template to accommodate bazel build as part of it, but this seems not as clean as template inheritance with jinja.  

Here is a link to the latest bazel run with this change https://github.com/pytorch/pytorch/actions/runs/1004656584

Differential Revision: [D29562260](https://our.internmc.facebook.com/intern/diff/D29562260)

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: .github/templates/bazel_ci_workflow.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+          fi
+          echo ::set-output name=rebuild::yes
+      - name: Build and push docker image
+        if: steps.check.outputs.rebuild
+        env:
+          DOCKER_TAG: ${{ steps.calculate-tag.outputs.docker_tag }}
+          DOCKER_SKIP_S3_UPLOAD: 1
+        run: |
+          export IMAGE_NAME=${DOCKER_IMAGE_BASE#308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/}
+          cd .circleci/docker && ./build_docker.sh
```
Comment (malfet) [inline .github/templates/bazel_ci_workflow.yml.j2:98]: This section seems to be word-by-word copy of one from linux_ci_workflow.yml.j2 , see small snippet below 
https://github.com/pytorch/pytorch/blob/864dcbb2cc65352a9ba260c3cf05ee30f4bcc23d/.github/templates/linux_ci_workflow.yml.j2#L36-L40

I wonder if we can modify our template generating system to merge those two somehow

Comment (rsemenov) [reply .github/templates/bazel_ci_workflow.yml.j2:98]: I mentioned this in the PR description, there is inheritance concept with jinya that I think we should start using https://jinja.palletsprojects.com/en/3.0.x/templates/#template-inheritance . I will try it and see how much change that would be to fit in this PR.

Comment (rsemenov) [reply .github/templates/bazel_ci_workflow.yml.j2]: Updated to use inheritance. Going forward we should start using [macros](https://jinja.palletsprojects.com/en/3.0.x/templates/#macros). They act like functions and are great for reusability. With macros we potentially could reuse `render_test_results` with passing in some parameters. But this introduces unnecessary complexity at the moment.

**Comment by rsemenov** [comment]
@rsemenov has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29562260).

**Pull request closed without merging**